File: domain/src/main/java/com/ahn/domain/currency/usecase/SaveExchangeFromCurrencyUseCase.kt
package com.ahn.domain.currency.usecase

import com.ahn.domain.currency.repository.CurrencySelectionRepository
import javax.inject.Inject

class SaveExchangeFromCurrencyUseCase @Inject constructor(private val repository: CurrencySelectionRepository) {
    suspend operator fun invoke(code: String) {
        repository.saveExchangeFromCurrencyCode(code)
    }
}

File: ios/Packages/Domain/Sources/Domain/Currency/UseCase/SaveExchangeFromCurrencyUseCase.swift
public struct SaveExchangeFromCurrencyUseCase: Sendable {
    private let repository: any CurrencySelectionRepository

    public init(repository: any CurrencySelectionRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ code: String) async throws {
        try await repository.saveExchangeFromCurrencyCode(code)
    }
}
